lib/private/Http/CookieHelper.php

```diff
@@ -0,0 +1,195 @@
+	public const HTTPONLY = 'httponly';
+	public const SAMESITE = 'samesite';
+	public const POLYFILL_DETECT = 'detect';
+	public const POLYFILL_FALLBACK = 'fallback';
+
+	public static function setCookie(string $name, string $value = '', array $options = [], string $polyfill = self::POLYFILL_FALLBACK): void {
+		// check if given cookie options are allowed
+		foreach (array_keys($options) as $option) {
+			if (!in_array($option, array(self::PATH, self::DOMAIN, self::EXPIRES, self::SECURE, self::HTTPONLY, self::SAMESITE))) {
+				trigger_error(
+					'unknown cookie option : ' . $option,
+					E_WARNING
+				);
+			}
+		}
+
+		// check if given polyfill mode is supported
+		if (!in_array($polyfill, array(self::POLYFILL_DETECT, self::POLYFILL_FALLBACK))) {
+			trigger_error(
+				'unknown cookie polyfill : ' . $polyfill,
+				E_WARNING
+			);
+		}
+
+		// defaults
+		$path = array_key_exists(self::PATH, $options) ? $options[self::PATH] : '';
+		$domain = array_key_exists(self::DOMAIN, $options) ? $options[self::DOMAIN] : '';
+		$expires = array_key_exists(self::EXPIRES, $options) ? $options[self::EXPIRES] : 0;
+		$secure = array_key_exists(self::SECURE, $options) ? $options[self::SECURE] : false;
+		$httponly = array_key_exists(self::HTTPONLY, $options) ? $options[self::HTTPONLY] : false;
+		$sameSite = array_key_exists(self::SAMESITE, $options) ? $options[self::SAMESITE] : '';
+		$header = array(sprintf('Set-Cookie: %s=%s', $name, rawurlencode($value)));
+
+		// disable samesite
+		// if samesite is set to none secure is required
+		// if polyfill is set to POLYFILL_DETECT and $user_agent does not support samesite
+		if (
+			$sameSite === self::SAMESITE_NONE && $secure !== true ||
+			$polyfill === self::POLYFILL_DETECT && !self::canSameSite()
+		) {
+			$sameSite = '';
+		}
+
+		// create a second cookie to polyfill unsupported browsers
+		// if set to POLYFILL_FALLBACK use CookieHelper::getRequestCookie(...)
+		// https://web.dev/samesite-cookie-recipes/#handling-incompatible-clients
+		if ($polyfill === self::POLYFILL_FALLBACK && $sameSite !== '') {
+			unset($options[self::SAMESITE]);
+			self::setCookie(self::legacyName($name), $value, $options, self::POLYFILL_DETECT);
+		}
+
+		if ($path !== '') {
+			array_push($header, sprintf('Path=%s', $path));
+		}
+
+		if ($domain !== '') {
+			array_push($header, sprintf('Domain=%s', $domain));
+		}
+
+		if ($expires > 0) {
+			array_push($header, sprintf('Max-Age=%d', $expires));
+		}
+
+		if ($secure === true) {
+			array_push($header, 'Secure');
+		} else {
+			$sameSite = '';
+		}
+
+		if ($httponly) {
+			array_push($header, 'HttpOnly');
+		}
+
+		foreach (array(self::SAMESITE_NONE, self::SAMESITE_LAX, self::SAMESITE_STRICT) as $v) {
+			if (strtolower($sameSite) === strtolower($v)) {
+				array_push($header, sprintf('SameSite=%s', $v));
+				break;
+			}
+		}
+
+		header(implode('; ', $header), false);
+	}
+
+	// if polyfill is set to true it returns the cookie legacy name
```

Comment:
```suggestion
	// return the cookie legacy name
```

I don't see an "if" statement in this function.